packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+    isSpaceLengthHandled = true;
+  }
+  rangeEnd = oldPlainText.indexOf(' ', oldPlainTextEndIndex);
+  if (rangeEnd === -1 || rangeEnd === undefined) {
+    // check if space symbol is not found
+    rangeEnd = plainTextEndIndex;
+  } else if (!isSpaceLengthHandled) {
+    // +1 to include the space symbol
+    rangeEnd += 1;
+  }
+  isSpaceLengthHandled = true;
+
+  if (rangeStart === -1 || rangeStart === undefined || rangeStart < tag.plainTextBeginIndex) {
+    // rangeStart should be at least equal to tag.plainTextBeginIndex
+    rangeStart = tag.plainTextBeginIndex;
+  }
+  if (rangeEnd > plainTextEndIndex) {
+    // rangeEnd should be at most equal to plainTextEndIndex
+    rangeEnd = plainTextEndIndex;
+  }
+  if (rangeStart === tag.plainTextBeginIndex && rangeEnd === plainTextEndIndex) {
+    // the whole tag should be removed
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
+    plainTextSelectionEndIndex = tag.plainTextBeginIndex + processedChange.length;
+    processedChange = '';
+    lastProcessedHTMLIndex = closeTagIndex + closeTagLength;
+  } else {
+    // only part of the tag should be removed
+    let startChangeDiff = 0;
+    let endChangeDiff = 0;
+    // need to check only rangeStart > tag.plainTextBeginIndex as when rangeStart === tag.plainTextBeginIndex startChangeDiff = 0 and mentionTagLength shouldn't be subtracted
+    if (rangeStart > tag.plainTextBeginIndex) {
+      startChangeDiff = rangeStart - tag.plainTextBeginIndex - mentionTagLength;
+    }
+    endChangeDiff = rangeEnd - tag.plainTextBeginIndex - mentionTagLength;
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex + tag.openTagBody.length + startChangeDiff);
+
+    if (startIndex < tag.plainTextBeginIndex) {
+      // if the change is before the tag, the selection should start from startIndex (rangeStart will be equal to tag.plainTextBeginIndex)
+      plainTextSelectionEndIndex = startIndex + change.length;
+    } else {
+      // if the change is inside the tag, the selection should start with rangeStart
+      plainTextSelectionEndIndex = rangeStart + processedChange.length;
+    }
+    lastProcessedHTMLIndex = tag.openTagIndex + tag.openTagBody.length + endChangeDiff;
+    // processed change should not be changed as it should be added after the tag
+  }
+  return { result, updatedChange: processedChange, htmlIndex: lastProcessedHTMLIndex, plainTextSelectionEndIndex };
+};
+
+/**
+ * Closing tag information
+ *
+ * @private
+ */
+export type ClosingTagInfoResult = {
+  plainTextEndIndex: number;
+  closeTagIndex: number;
+  closeTagLength: number;
+};
+
+/**
+ * Get closing tag information if exists otherwise return information as for self closing tag
+ *
+ * @private
+ * @param tag - Tag data.
+ * @returns Closing tag information for the provided tag.
+ */
+export const getTagClosingTagInfo = (tag: TagData): ClosingTagInfoResult => {
```

Comment:
`getClosingTag` ?

Reply:
> 

I think it make sense to keep the info part since what we get from this function is the information of a closing tag (plainTextEndIndex: number; closeTagIndex: number; closeTagLength: number) and not the closing tag itself?
Or maybe we can rename it to getClosingTagData to align with TagData type?